src/strategies/dfyn-staked-in-farms/index.ts

```diff
@@ -0,0 +1,145 @@
+import { formatUnits } from '@ethersproject/units';
+import { multicall } from '../../utils';
+import { subgraphRequest } from '../../utils';
+
+export const author = 'vatsal';
+export const version = '0.1.0';
+
+
+const DFYN_SUBGRAPH_URL = {
+  '137': 'https://api.thegraph.com/subgraphs/name/ss-sonic/dfyn-v5'
+};
+
+const getAllLP = async (skip, stakedLP, network) => {
+  let params =
+  {
+    pairs: {
+      __args: {
+        where: {
+          id_in: stakedLP.map((address) => address.toLowerCase())
+        },
+        skip: skip,
+        first: 1000
+      },
+      id: true,
+      reserve0: true,
+      reserve1: true,
+      totalSupply: true,
+      token0: {
+        id: true,
+        symbol: true,
+      },
+      token1: {
+        id: true,
+        symbol: true,
+      }
+    }
+  }
+
+  try {
+    const response = await subgraphRequest(DFYN_SUBGRAPH_URL[network], params)
+    return response.pairs;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+const getLP = async (stakedLP, network) => {
+  let lp = []
+  let results = []
+  do {
+    results = await getAllLP(lp.length, stakedLP, network);
+    lp = lp.concat(results)
+  } while (results.length === 1000);
```

Comment:
if `results` returned by subgraph are not equal to 1000, this will be a never-ending loop right? maybe you mean to add `lp.length < 1000`? Let me know if i am thinking in wrong way

Reply:
On the contrary, loop will only continue if `results.length === 1000`, otherwise it will break. Since subgraph returns at max 1000 items in one call, I am checking if `results.length === 1000`. If it is, then it needs to fetch the items after those 1000 items. If less than 1000 items are returned, the loop will break since `results.length` will not be equal to 1000. Either way, this piece of code has been included to make the strategy future proof. Currently there are only around 10 Dfyn farms so only 10 liquidity pairs are returned.

Reply:
oh now I get it, I was overlooking :)